connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -19,24 +19,25 @@
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.kafka.common.utils.Exit;
+import org.apache.kafka.common.utils.Utils;
 import org.apache.kafka.connect.errors.ConnectException;
 import org.apache.kafka.connect.runtime.isolation.Plugins;
 import org.apache.kafka.connect.runtime.rest.entities.ActiveTopicsInfo;
 import org.apache.kafka.connect.runtime.rest.entities.ConfigInfos;
-import org.apache.kafka.connect.runtime.rest.entities.ConnectorOffsets;
 import org.apache.kafka.connect.runtime.rest.entities.ConnectorInfo;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorOffsets;
 import org.apache.kafka.connect.runtime.rest.entities.ConnectorStateInfo;
 import org.apache.kafka.connect.runtime.rest.entities.ServerInfo;
 import org.apache.kafka.connect.runtime.rest.errors.ConnectRestException;
+import org.eclipse.jetty.client.HttpClient;
```

Comment:
Yeah, I did want to avoid doing this, but it was essentially a last resort. The `HttpURLConnection` API doesn't support the `PATCH` HTTP method and there is a won't fix bug ticket in OpenJDK for this - https://bugs.openjdk.org/browse/JDK-7016595. The new `HttpClient` (incubated in Java 9 and made official in Java 11) which supersedes `HttpURLConnection` does fix this issue, but since we still support Java 8, that wasn't an option either. Since the Jetty HTTP client is already being used by the main Connect codebase, I figured it'd be the best option here.

Reply:
Thanks for the clarification, I think it's reasonable to use jetty in that circumstance.